solidity/ERC20ForSPL/ERC20ForSPLFactory.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import "./openzeppelin-fork/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "./openzeppelin-fork/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "./openzeppelin-fork/contracts/proxy/beacon/BeaconProxy.sol";
+import "./ERC20ForSPL.sol";
+
+
+/// @title ERC20ForSPLFactory
+/// @author https://twitter.com/mnedelchev_
+/// @notice This contract serve as a factory to deploy interface contracts of already deployed SPLToken on Solana.
+/// @dev This contract is built with forked OpenZeppelin's UUPS standard and it's a Beacon contract at the same time. The storage is defined in the following way:
+/// @dev Storage slot 0 - taken by the forked BeaconProxy's implementation.
+/// @dev Storage slot 1 - taken by the forked UUPS's implementation.
+/// @dev Storage slot 2 - taken by the forked OwnableUpgradeable's owner.
+/// @dev Every next slot is defined by the needs of the ERC20ForSPLFactory.
+/// @custom:oz-upgrades-unsafe-allow constructor
+contract ERC20ForSPLFactory is OwnableUpgradeable, UUPSUpgradeable {
```

Comment:
Do I correctly understand that `_implementation` is about ERC20ForSpl implementation and `_uupsImplementation` is about current proxy implementation? Which way is it possible to update _uupsImplementation?

Reply:
Correct, `_implementation` is the Beacon Proxy implementations and the ERC20ForSPLs contracts are the Beacon Proxies. This implementation can by upgraded by using method `upgradeTo` in the **[Factory](https://github.com/neonlabsorg/neon-evm/blob/update/ERC20ForSPL-and-ERC20ForSPLMintable/solidity/ERC20ForSPL/ERC20ForSPLFactory.sol#L75)**.

`_uupsImplementation` is the UUPS implementation of the factory it self, because the DAO can update the factory code too. This implementation can be upgraded by using method `upgradeToAndCall` which is a method that factory inherit from **[UUPSUpgradeable lib](https://github.com/neonlabsorg/neon-evm/blob/update/ERC20ForSPL-and-ERC20ForSPLMintable/solidity/ERC20ForSPL/openzeppelin-fork/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol#L92)**.

Reply:
Got it.